contracts/HeapOrdering.sol

```diff
@@ -174,15 +175,16 @@ library HeapOrdering {
         if (_id == address(0)) revert AddressIsZero();
 
         // Put the account at the end of accounts.
-        uint256 accountsLength = _heap.accounts.length;
-        _heap.accounts.push(Account(_id, _value));
+        uint256 accountsLength = _heap.arrayLength;
+        _heap.accounts[accountsLength] = Account(_id, _value);
+        _heap.arrayLength = uint128(accountsLength + 1);
```

Comment:
I guess it's fine to not use SafeCast because these cannot realistically overflow?